From 7a8ec129fb54f188b2da6588c9d24d3a36eb0d39 Mon Sep 17 00:00:00 2001
From: Alan Guo Xiang Tan <gxtan1990@gmail.com>
Date: Fri, 17 Dec 2021 14:24:32 +0800
Subject: [PATCH] SECURITY: Disable MessageBus::Diagnostics.

MessageBus::Diagnostics allows anyone with access to carry out certain
operations that may result in a denial of service. The impact of this is
greater on multisiite clusters.
---
 config/initializers/004-message_bus.rb | 1 -
 1 file changed, 1 deletion(-)

diff --git a/config/initializers/004-message_bus.rb b/config/initializers/004-message_bus.rb
index a566d4658f57..87fa9c4c7613 100644
--- a/config/initializers/004-message_bus.rb
+++ b/config/initializers/004-message_bus.rb
@@ -120,7 +120,6 @@ def setup_message_bus_env(env)
 MessageBus.long_polling_enabled = SiteSetting.enable_long_polling
 MessageBus.long_polling_interval = SiteSetting.long_polling_interval
 MessageBus.cache_assets = !Rails.env.development?
-MessageBus.enable_diagnostics
 
 if Rails.env == "test" || $0 =~ /rake$/
   # disable keepalive in testing
